packages/@react-spectrum/label/src/HelpText.tsx

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2021 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {classNames, useDOMRef, useStyleProps} from '@react-spectrum/utils';
+import {DOMProps, DOMRef, SpectrumHelpTextProps, StyleProps} from '@react-types/shared';
+import {filterDOMProps} from '@react-aria/utils';
+import React, {HTMLAttributes} from 'react';
+
+interface HelpTextProps extends SpectrumHelpTextProps {
```

Comment:
Is there a better way to have HelpText accept `descriptionProps` and `errorMessageProps` than creating another `HelpTextProps` interface here? The main one lives in [inputs.d.ts](https://github.com/adobe/react-spectrum/pull/1846/files#diff-e555c507eaa782d07e38010e82ab58729e31e968c9d12d3cc78ec9d9fa962a9cR64). I was thinking of adding these 2 props to `SpectrumHelpTextProps`, but I think I shouldn't because that’s the interface that components like Combobox will extend.

Reply:
What you have here seems reasonable, it is similar to how we defined `TextFieldBaseProps` for TextFieldBase where we extended `SpectrumTextFieldProps` and then manually defined things like `labelProps` and `inputProps`.

Should those props be required? Perhaps they shouldn't be since errorMessage and description may not be provided right?

Reply:
I was wondering about making them optional or required. Because of this line I decided to make them required, but I want to look into making them optional, like [labelProps](https://github.com/adobe/react-spectrum/blob/main/packages/@react-spectrum/textfield/src/TextFieldBase.tsx#L37) is in TextFieldBase.
> descriptionProps and errorMessageProps will need to be added to the return types for all of these hooks, and passed through to the Field component in Spectrum.

Reply:
Sounds good, we can have others weigh in as well.

Reply:
would it make more sense for message Props to be propagated through slots into here?
I'm assuming they are needed to pass classNames/ids mostly? and I saw you used `useSlotId` earlier

Reply:
Hey @snowystinger, could you give me an example of how to propagate the message props using slots here? Sorry I don't totally understand. As of now, yes `descriptionProps` and `errorMessageProps` only contain an `id`:

https://github.com/adobe/react-spectrum/pull/1846/files#diff-012e9e517a76baacb63e3e4d93716f8d0acd4dadf5477073cae223ca3832e9d0R50-R55